### PR TITLE
test: make all the tests run in serial

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Problem Summary: 

For now, the CI is extremely unstable because some parallel tests access the global variable. Since there are too many of such tests, we try to make all the tests run in serial first, and then parallelize them later as an optimization. 

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: br/pkg/lightning/backend/backend_test.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ func (s *backendSuite) tearDownTest() {
 }
 
 func TestOpenCloseImportCleanUpEngine(t *testing.T) {
-	t.Parallel()
+
```
Comment (wjhuang2016) [inline br/pkg/lightning/backend/backend_test.go:42]: Redundant empty line.

File: br/pkg/lightning/backend/backend_test.go [hunk 2]
```diff
@@ -72,7 +72,7 @@ func TestOpenCloseImportCleanUpEngine(t *testing.T) {
 }
 
 func TestUnsafeCloseEngine(t *testing.T) {
-	t.Parallel()
+
```
Comment (wjhuang2016) [inline br/pkg/lightning/backend/backend_test.go:75]: Here

**Review by wjhuang2016 — COMMENTED** [review]
I think it's should be 0 line added.

**Review by wjhuang2016 — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Review by disksing — APPROVED** [review]
TLDR

**Review by jackysp — APPROVED** [review]

**Review by XuHuaiyu — APPROVED** [review]

**Comment by bb7133** [comment]
Le'ts keeps the an eyes on https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/activity and check the estimated time of unit_test before/after this PR is merged.

**Comment by tisonkun** [comment]
Two suggestions:

1. These changes break the convention of separating `_serial_test.go` and `_test.go`. You will create a new confusion that why there is this difference but actually all tests run in serial.
2. Do you have an analysis about where the unstable tests located? I don't think a global find and replace a good choice because it can "resolve" non-existing "issues". Basically, I suspect that tests under the following packages, due to code quality, can be hardly run in parallel:
    * ddl
    * server
    * session
    * planner
    * executor
    * expression
... while others aren't so much "unstable".

**Comment by tangenta** [comment]
@tisonkun

1. For the confusion about "_serial_test.go", we can merge them back together.
2. One of the purposes of this PR is to prevent the fixes like "change TestXXX from parallel to serial" in the future. It is a waste of effort to serialize the tests because the improvement is not significant (13min -> 9min).
    
    To speed up the test, we can put the parallel mechanism to a higher level. For example, split them into a few packages and run them in different pods.

**Comment by tisonkun** [comment]
> 1. For the confusion about "_serial_test.go", we can merge them back together.

It's OK if you wish.

> 2. ...

I do never insist for performance topic especially I experienced multiple projects run tests in hours. It's about resolving non-issues but not a blocker. Do what you like.

**Comment by bb7133** [comment]
Some results about the execution tie of unit-test from the current CI:

This PR:
* [No.24713](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/24713/pipeline/) - 10m28s
* [No.24700](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/24700/pipeline) - 11m2s
* [No.24698](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/24698/pipeline/) - 13m58s
* [No.24691](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/24691/pipeline/) - 16m36s+
* [No.24686](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/24686/pipeline/) - 16m2s+
* [No.24671](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/24671/pipeline) - 13m19s
* [No.24639](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/24639/pipeline) - 13m2s

Before:
* [No.24642](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/24642/pipeline/) - 10m5s
* [No.24633](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/24633/pipeline) - 9m1s
* [No.24619](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/24619/pipeline) - 8m59s
* [No.24613](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/24613/pipeline) - 9m56s
* [No.24610](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/24610/pipeline) - 10m47s
* [No.24596](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/24596/pipeline) - 9m6s
* [No.24582](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/24582/pipeline) - 9m18s

For [No.24686](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/24686/pipeline/) and [No.24671](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/24671/pipeline), it time-outed because the previous limit for single package is 10min but I think it is close to finish or we have some other unknown issues to solve.

In total, the estimated increase of CI time is about about 50%(~9min -> ~13min) but I think this is acceptable since the tests can be more stable.

**Review by bb7133 — APPROVED** [review]

**Comment by tangenta** [comment]
I don‘t see where there is inappropriate to "resolve non-issues". They are just unrelated harmless changes.
